### DISCUSSION TITLE
Need help "please plug in device"

### DISCUSSION THREAD

**Discussion by BobbyBoyGaming** [micronucleus/micronucleus#235] [Q&A]
![please plug in device](https://user-images.githubusercontent.com/46551829/121764886-40fce780-cb15-11eb-9286-1c7efa7798d0.jpg)
hello I need help with the xd002, I created a keymap using QMK but this ATTiny MCU 2-button keyboard can only be flashed via the "micronucleus CLI exe file" found in https://github.com/micronucleus/micronucleus

I created a keymap, of the appropriate size, according to QMKSYS, then I installed the zadig drivers according to the micronucleus github instructions (windows 10). then I start windows CMD as admin and cd into the micronucleus exe and enter the command 'micronucleus --run <firmware_file_name>

But I keep getting "please plug in the device"
I tried unplugging and plugging (also with the reset feature, while holding the button nearest to microUSB socket while plugging in the PCB)

I tried doing the "micronucleus --erase-only" command first but I still get "please plug in device..."

Does anybody know what is going on? I only have Windows 10 at the moment.

**Comment by cpldcpu** [comment]
It looks like micronucleus is not active on the device your try to program. 

I have no idea what QMKSYS is, and which version of Micronucleus or invokation scheme it uses. Have you already inquired there?

**Comment by BobbyBoyGaming** [comment]
These are the instructions that came with the QMK firmware program, Could you please read them and tell me if they look correct? I don't have any programming experience, I can't tell if these instructions say how to put the bootloader into the keypad. Can you please tell me if you think these instructions make any sense? or are they missing the step on how to put the micronucleus into the MCU of the keypad?
- - - - - - -
# xd002

![xd002](https://cdn.shopify.com/s/files/1/2711/4238/products/xd002_1024x1024.jpg?v=1522134411)

2% Custom mechanical keyboard. ATtiny85 powered, with 2*WS2812 LEDs, and the micronucleus bootloader.

**Note**: Due to limited firmware space, a _**lot**_ of features have to be disabled to get a functioning QMK based keyboard.

* Keyboard Maintainer: [zvecr](https://github.com/zvecr)
* Hardware Supported: xd002
* Hardware Availability: [kprepublic](https://kprepublic.com/products/xd002-xiudi-2-custom-mechanical-keyboard-2-keys-underglow-and-switch-rgb-pcb-programmed-hot-swappable-macro-key-aluminum-case)

Make example for this keyboard (after setting up your build environment):

    make xd002:default

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).

## Flashing
### Prerequisites

```bash
git clone https://github.com/micronucleus/micronucleus.git
cd micronucleus/commandline/
sudo make install
```

On Linux, you’ll need proper privileges to access the MCU. You can either use sudo when flashing firmware, or place [these files](https://github.com/micronucleus/micronucleus/blob/master/commandline/49-micronucleus.rules) in /etc/udev/rules.d/. Once added run the following:

```bash
sudo udevadm control --reload-rules
sudo udevadm trigger
```

### Instructions

**Reset Key**: Hold down key nearest to the USB socket while plugging in the keyboard.

```bash
make xd002:default:flash

# or directly with...
micronucleus --run <firmware.hex>
```

### Recovery

* [Original Firmware](https://github.com/xiudi/Attiny85_vusb_pad_test)
* [Bootloader Repair](https://digistump.com/wiki/digispark/tutorials/proisp)

**Reply by ArminJo** [reply]
>  Could you please read them and tell me if they look correct? I don't have any programming experience,

Very nice. What are you willing to pay for 😀 ?
